### PR TITLE
ADD: Toggle for disabling Tor daemon

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/1wm902ayp9f2zg5ggeeb1q67nr?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8428)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

File: blue_modules/BlueElectrum.js [hunk 1]
```diff
@@ -90,10 +90,30 @@ async function isDisabled() {
...
+    if (savedValue === null) {
+      isTorDaemonDisabled = false;
+    } else {
+      isTorDaemonDisabled = savedValue;
+    }
+  } catch {
+    isTorDaemonDisabled = true;
+  }
+
+  return !!isTorDaemonDisabled;
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.js:106]: maybe 
`return !!isTorDaemonDisabled || !isTorCapable;`
?

File: class/wallets/lightning-custodian-wallet.js [hunk 2]
```diff
@@ -72,12 +72,13 @@ export class LightningCustodianWallet extends LegacyWallet {
...
       baseURI: this.baseURI,
     });
-
-    if (isTorCapable && this.baseURI && this.baseURI?.indexOf('.onion') !== -1) {
-      this._api = new torrific.Torsbee({
-        baseURI: this.baseURI,
-      });
-    }
+    BlueElectrum.isTorDaemonDisabled().then(isDisabled => {
+      if (!isDisabled && this.baseURI && this.baseURI?.indexOf('.onion') !== -1) {
```
Comment (Overtorment) [inline class/wallets/lightning-custodian-wallet.js:76]: i hope this wont cause issues as its an async call. hope we don't have immediate network wetch from lndhub _right_ after init()

File: blue_modules/BlueElectrum.js [hunk 3]
```diff
@@ -90,10 +90,30 @@ async function isDisabled() {
   return !!isDisabled;
 }
 
+async function isTorDaemonDisabled() {
```
Comment (Overtorment) [inline blue_modules/BlueElectrum.js:93]: this belongs in `'./environment'`

**Review by Overtorment — CHANGES_REQUESTED** [review]

File: blue_modules/BlueElectrum.js (hunk 1 above)
Comment (marcosrdz) [reply blue_modules/BlueElectrum.js:106]: Two different purposes.

File: class/wallets/lightning-custodian-wallet.js (hunk 2 above)
Comment (marcosrdz) [reply class/wallets/lightning-custodian-wallet.js:76]: Is there a way to refactor in a way that avoids race condition?

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/fyu5yd5emzrcnjtxj1vxa04c08?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8430)

File: class/wallets/lightning-custodian-wallet.js (hunk 2 above)
Comment (Overtorment) [reply class/wallets/lightning-custodian-wallet.js:75]: cant think of anything.
afair we have init() only on startup and import

**Review by Overtorment — APPROVED** [review]

**Pull request merged**